app/src/main/kotlin/com/mgorski/discountasciiwarehouse/network/AsciiItemsProvider.kt

```diff
@@ -0,0 +1,11 @@
+package com.mgorski.discountasciiwarehouse.network
+
+import com.mgorski.discountasciiwarehouse.model.toAsciiItem
+import rx.schedulers.Schedulers
+
+class AsciiItemsProvider(private val service: AsciiWarehouseService) {
+    fun getAsciiItems(limit: Int? = null, skip: Int? = null, tagsQuery: String? = null, onlyInStock: Int? = null) =
+            service.getAsciiItems(limit, skip, tagsQuery, onlyInStock)
+                    .subscribeOn(Schedulers.io())

```

Comment:
Consider moving this delegation to io thread directly to the calling place. This way you will have more fine grained control over where it will be run.